src/core/ext/xds/xds_client.cc

```diff
@@ -1316,8 +1319,15 @@ void XdsClient::ChannelState::AdsCallState::OnStatusReceivedLocked(
     // Try to restart the call.
     parent_->OnCallFinishedLocked();
     // Send error to all watchers.
-    xds_client()->NotifyOnErrorLocked(
-        GRPC_ERROR_CREATE_FROM_STATIC_STRING("xds call failed"));
+    char* status_details = grpc_slice_to_c_string(status_details_);
+    xds_client()->NotifyOnErrorLocked(GRPC_ERROR_CREATE_FROM_COPIED_STRING(
+        absl::StrFormat(
+            "xDS call failed: xDS server: %s, ADS call status code=%d, "
+            "details='%s', error='%s'",
+            chand()->server_.server_uri.c_str(), status_code_, status_details,
```

Comment:
`absl::StrFormat()` knows how to handle `std::string` arguments, so no need to call `.c_str()` on any parameters.

Reply:
done